mlflow/tracking/client.py

```diff
@@ -1536,16 +1555,52 @@ def log_image(
                 "`key` to log dynamic image charts or `artifact_file` for saving static images. "
             )
 
+        import numpy as np
+
+        # Convert image type to PIL if its a numpy array
+        if isinstance(image, np.ndarray):
+            image = convert_to_pil_image(image)
+        elif isinstance(image, Image):
+            image = image.to_pil()
+        else:
+            # Import PIL and check if the image is a PIL image
+            import PIL
+
+            if not isinstance(image, PIL.Image.Image):
+                raise TypeError("Unsupported image object type.")
```

Comment:
A general advice on error message: we should tell users 1) what's going wrong 2) what user is feeding in 3) what is allowed.  Something like the following would be better
```
if not isinstance(image, PIL.Image.Image):
    raise TypeError(
        f"Unsupported image object type: {type(image)}, `image` input of "
        "`mlflow.log_image` must be one of ..."
    )
```